src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -2083,87 +2085,114 @@ void InstrumentClipView::adjustVelocity(int32_t velocityChange) {
 				Note* note = noteRow->notes.getElement(noteI);
 				while (note && note->pos - editPadPresses[i].intendedPos < editPadPresses[i].intendedLength) {
 					if (display->hasPopup()) {
+						// Sean: check for pop-up so that you don't change encoder turn (cause you may just want to see
+						// the value)
 						noteRow->changeNotesAcrossAllScreens(note->pos, modelStackWithNoteRow, action,
 						                                     CORRESPONDING_NOTES_ADJUST_VELOCITY, velocityChange);
 					}
 
-					if (velocityValue == 0) {
-						velocityValue = note->getVelocity();
-					}
-					else {
-						if (velocityValue != note->getVelocity()) {
-							velocityValue = 255; // Means "multiple"
-						}
-					}
+					getVelocityValue(velocityValue, note->getVelocity());
+
 					numNotesThisSquare++;
 					velocitySumThisSquare += note->getVelocity();
 
 					noteI++;
 					note = noteRow->notes.getElement(noteI);
 				}
 
-				editPadPresses[i].intendedVelocity =
-				    velocitySumThisSquare / numNotesThisSquare; // Get the average. Ideally we'd have done this when
-				                                                // first selecting the note too, but I didn't
+				// Rohan: Get the average. Ideally we'd have done this when first selecting the note too, but I didn't
+
+				// Sean: not sure how getting the average when first selecting the note would help because the average
```

Comment:
I think this is "when first selecting the note" like when you first press it, before starting to adjust velocity. That first press auditions at the velocity of the first note in the square

Reply:
Just remove the comments though since I don't think it matters